rollout/sync.go

```diff
@@ -584,7 +584,19 @@ func (c *rolloutContext) calculateRolloutConditions(newStatus v1alpha1.RolloutSt
 	} else {
 		if completeCond != nil {
 			updateCompletedCond := conditions.NewRolloutCondition(v1alpha1.RolloutCompleted, corev1.ConditionFalse, conditions.RolloutCompletedReason, conditions.RolloutCompletedReason)
-			conditions.SetRolloutCondition(&newStatus, *updateCompletedCond)
+			changed := conditions.SetRolloutCondition(&newStatus, *updateCompletedCond)
+
+			// if any rs status changes (e.g., pod restarted, evicted -> recreated ) to a previous completed rollout,
+			// we need to reset the progressCondition to avoid timeout
```

Comment:
```suggestion
			// If the ReplicaSet status changes (e.g., one of the pod restarts, evicted -> recreated) for a previously
			// completed rollout, we'll need to reset the rollout's condition to `PROGRESSING` to avoid any timeouts.
```

**Minor:** Making the comment a little bit more clear for any future maintainers who are not very familiar to this logic.